_posts/2015-12-18-Rails-5-0-beta1.markdown

```diff
@@ -0,0 +1,88 @@
+
+## Action Cable
+The big new thing in Rails 5.0 is a brand-new framework for handling WebSockets called [Action Cable][actioncable]. It's a completely integrated solution that includes an EventMachine-powered connection loop, a thread-backed channels layer for server-side processing, and a JavaScript layer for client-side interaction. It's incredibly easy to use, and makes designing live features like chat, notifications, and presence so much easier. It's what's powering all those features of [Basecamp 3][basecamp], if you want to see it in action.
+
+What's really lovely about Action Cable is that you get access to your entire Active Record and PORO domain model in your WebSockets work. We even added a brand-new `ActionController::Renderer` system that makes it trivial to render your templates outside of controllers, when you want to reuse server-side templates for WebSocket responses.
+
+In development, Action Cable runs in-process with the rest of your app. To do this, we've switched the default development server from Webrick to Puma. In production, you may well want to run Action Cable servers in their own processes. That's how we run it at Basecamp at scale.
+
+Special thanks to [Pratik Naik][pratiknaik] and [Javan Makhmali][javanmakhmali] for their formative work.
+
+## API mode
+Rails is not only a great choice when you want to build a full-stack application that uses server-side rendering of HTML templates, but also a great companion for the new crop of client-side JavaScript or native applications that just needs the backend to speak JSON. We've made this even clearer now with the new --api mode. If you create a new Rails application using `rails new backend --api`, you'll get a slimmed down skeleton and configuration that assumes you'll be working with JSON, not HTML.
+
+There's still more work to be done on this feature, but we're off to a great start. By default, API mode just relies on #to_json calls on model classes. But you can either use [Jbuilder][jbuilder], [Active Model Serializers][ams], or look at the new [JSONAPI::Resources][jsonapi] project for a more advanced solution.
+
+Thanks in particular to [Santiago Pastorino][santiagopastorino] and [Jorge Bejar][jorgebejar] for making this happen.
+
+## One Rails command to rule them all
+Why are some commands living in `bin/rails` and some commands living in `bin/rake`? That's a common question, especially for beginners, and we never had a good answer (just lots of technical excuses). So now we've committed to making `bin/rails` the one master command to rule them all. All your rake commands are available through here as a gateway, but we'll eventually port many of them over. So your fingers will now have to get used to `bin/rails db:migrate` instead of `bin/rake db:migrate`. That should only take a few months!
+
+[Kasper Timm Hansen][kasper] has been herding this project.
```

Comment:
I think that work is awesome, but most people aren't going to be interfacing with it directly. This literally changes how everyone interfaces with Rails all the time. There are tons of very important fixes and improvements under the hood in Rails 5. This just calls out the most user-facing ones.